lighthouse-cli/test/smokehouse/lighthouse-runners/devtools.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview A runner that launches Chrome and executes Lighthouse via DevTools.
+ */
+
+import fs from 'fs';
+import os from 'os';
+import {spawn} from 'child_process';
+
+import {LH_ROOT} from '../../../../root.js';
+
+const devtoolsDir =
+  process.env.DEVTOOLS_PATH || `${LH_ROOT}/.tmp/chromium-web-tests/devtools/devtools-frontend`;
+
+/**
+ * @param {string} command
+ * @param {string[]} args
+ */
+async function spawnAndLog(command, args) {
+  let log = '';
+
+  /** @type {Promise<void>} */
+  const promise = new Promise((resolve, reject) => {
+    const spawnHandle = spawn(command, args);
+    spawnHandle.on('close', code => {
+      if (code === 0) resolve();
+      else reject(new Error(`Command exited with code ${code}`));
+    });
+    spawnHandle.on('error', reject);
+    spawnHandle.stdout.on('data', data => {
+      console.log(data.toString());
+      log += `STDOUT: ${data.toString()}`;
+    });
+    spawnHandle.stderr.on('data', data => {
+      console.log(data.toString());
+      log += `STDERR: ${data.toString()}`;
+    });
+  });
+  await promise;
+
+  return log;
+}
+
+/** @type {Promise<void>} */
+let beforeAllPromise;
+async function beforeAll() {
+  if (process.env.CI) return;
+
+  process.env.DEVTOOLS_PATH = devtoolsDir;
+  await spawnAndLog('bash', ['lighthouse-core/test/chromium-web-tests/download-devtools.sh']);
+  await spawnAndLog('bash', ['lighthouse-core/test/chromium-web-tests/roll-devtools.sh']);
+}
+
+/**
+ * Launch Chrome and do a full Lighthouse run via DevTools.
```

Comment:
I'm having trouble coming up with a non-awkward way to say this, but seems worth clarifying it's the current/local Lighthouse devtools bundle run in the latest DevTools frontend
vs
latest DevTools frontend with whatever Lighthouse version was last rolled into Chromium.